### PR TITLE
fix: add currentValue to yarn upgrade command

### PR DESCRIPTION
**Pull request by mmkal**
For `rangeStrategy: 'update-lockfile'`, replace `yarn upgrade some-dep`, which does nothing in some repos ([yarn bug](https://github.com/yarnpkg/yarn/issues/8236)), with `yarn upgrade some-dep@^1.2.3`. Where `^1.2.3` is the package.json range defined for dependency `some-dep`.

This is a workaround for a yarn bug, but barring unintended side-effects, it's a pretty reasonable one. For `update-lockfile`, the new command does actually more accurately represent what we want to do. Having said that, I have no idea what the unintended side-effects will be. I'm hoping CI will find them if they exist.

Came out of this issue originally: https://github.com/renovatebot/config-help/issues/752

### CODE REVIEW

File: lib/manager/npm/post-update/yarn.spec.ts [hunk 1]
```diff
@@ -94,6 +94,28 @@ describe(getName(__filename), () => {
       expect(fixSnapshots(execSnapshots)).toMatchSnapshot();
     }
   );
+  it.each([['1.22.0']])(
```
Comment (viceice) [inline lib/manager/npm/post-update/yarn.spec.ts:97]: please do not use `it.each` for a single value. It has no benefit and will slow down tests.

I know it's from copy'n'paste. 🙃

File: lib/manager/npm/post-update/yarn.ts [hunk 2]
```diff
@@ -94,7 +94,9 @@ export async function generateLockFile(
     // rangeStrategy = update-lockfile
     const lockUpdates = upgrades
       .filter((upgrade) => upgrade.isLockfileUpdate)
-      .map((upgrade) => upgrade.depName);
+      .map((upgrade) =>
+        [upgrade.depName, upgrade.currentValue].filter(Boolean).join('@')
+      );
```
Comment (viceice) [inline lib/manager/npm/post-update/yarn.ts:99]: ??? I think `currentValue` should never be `undefined`
```suggestion
      .map((upgrade) => `${upgrade.depName}@${upgrade.currentValue}`);
```

Does this work for older yarn versions too?

**Review by viceice — CHANGES_REQUESTED** [review]
@rarkins Do we really want this workaround? We want to support yarn v2 soon, so this would add another complexicy. 🤔

**Comment by rarkins** [comment]
@viceice I think this is a fine workaround, assuming it works and doesn't interfere with anything else.

File: lib/manager/npm/post-update/yarn.ts (hunk 2 above)
Comment (mmkal) [reply lib/manager/npm/post-update/yarn.ts:99]: :+1: will update.

Re older yarn versions, I checked on archive.org (!), and it's a documented way of using `yarn upgrade` for >3 years: https://web.archive.org/web/20170319085256/https://yarnpkg.com/lang/en/docs/cli/upgrade/

File: lib/manager/npm/post-update/yarn.spec.ts (hunk 1 above)
Comment (mmkal) [reply lib/manager/npm/post-update/yarn.spec.ts]: After removing the null check, there was nothing this test covered on top of the updated snapshot tests, so I removed it: https://github.com/renovatebot/renovate/pull/6759/commits/f31a7e7e35b005fda70fc2be1faae70c5632e8c0 - let me know if there's an additional test that should be included though.

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
@mmkal Did you test it against your real repo? Does it do what you expect?

**Comment by mmkal** [comment]
@viceice was just writing a message when I saw you merged - it _didn't_ work exactly the way I'd hoped!   It did create PRs, which is some progress, but the resolved version wasn't bumped in yarn.lock - only the matched semver range: https://github.com/mmkal/ts/pull/81/files. I would recommend this isn't released as-is.

**Pull request merged**